src/Paprika/Store/PagedDb.cs

```diff
@@ -357,7 +357,10 @@ private DbAddress SetNewRoot(RootPage root)
 
     private class ReadOnlyBatch(PagedDb db, RootPage root, string name) : IReportingReadOnlyBatch, IReadOnlyBatchContext
     {
-        private readonly ConcurrentDictionary<Keccak, uint> _idCache = new(Environment.ProcessorCount,
+        [ThreadStatic]
+        private static ConcurrentDictionary<Keccak, uint>? s_cache;
```

Comment:
I was thinking about caching it. Someday 😝

Reply:
225 sec trace

Before 126GB of ConcurrentDictionaries allocated from TxPool

<img width="510" alt="image" src="https://github.com/NethermindEth/Paprika/assets/1142958/ddf14acf-fb9d-4fa6-9f52-382b8001e483">

After 0 ConcurrentDictionaries allocated over same time period

<img width="519" alt="image" src="https://github.com/NethermindEth/Paprika/assets/1142958/18c43fce-e735-41d1-a082-a56852e01977">